components/prism-opencl.js

```diff
@@ -0,0 +1,49 @@
+(function (Prism) {
+	/* OpenCL kernel language */
+	Prism.languages.opencl = Prism.languages.extend('c', {
+		// Extracted from the official specs (2.0) and http://streamcomputing.eu/downloads/?opencl.lang (opencl-keywords, opencl-types) and http://sourceforge.net/tracker/?func=detail&aid=2957794&group_id=95717&atid=612384 (Words2, partly Words3)
+		'keyword': /\b(__attribute__|__constant|__global|__kernel|__local|__private|__read_only|__read_write|__write_only|_cl_command_queue|_cl_context|_cl_device_id|_cl_event|_cl_kernel|_cl_mem|_cl_platform_id|_cl_program|_cl_sampler|auto|bool|bool16|bool2|bool3|bool4|bool8|break|case|char|char16|char2|char3|char4|char8|cl_image_format|cl_mem_fence_flags|clk_event_t|complex|const|constant|continue|default|do|double|double16|double16x1|double16x16|double16x2|double16x4|double16x8|double1x1|double1x16|double1x2|double1x4|double1x8|double2|double2x1|double2x16|double2x2|double2x4|double2x8|double3|double4|double4x1|double4x16|double4x2|double4x4|double4x8|double8|double8x1|double8x16|double8x2|double8x4|double8x8|else|enum|event_t|extern|float|float16|float16x1|float16x16|float16x2|float16x4|float16x8|float1x1|float1x16|float1x2|float1x4|float1x8|float2|float2x1|float2x16|float2x2|float2x4|float2x8|float3|float4|float4x1|float4x16|float4x2|float4x4|float4x8|float8|float8x1|float8x16|float8x2|float8x4|float8x8|for|global|goto|half|half16|half2|half3|half4|half8|if|image1d_array_t|image1d_buffer_t|image1d_t|image2d_array_depth_t|image2d_array_msaa_depth_t|image2d_array_msaa_t|image2d_array_t|image2d_depth_t|image2d_msaa_depth_t|image2d_msaa_t|image2d_t|image3d_t|imaginary|inline|int|int16|int2|int3|int4|int8|intptr_t|kernel|local|long|long16|long2|long3|long4|long8|ndrange_t|packed|pipe|private|ptrdiff_t|quad|quad16|quad2|quad3|quad4|quad8|queue_t|read_only|read_write|register|reserve_id_t|restrict|return|sampler_t|short|short16|short2|short3|short4|short8|signed|size_t|sizeof|static|struct|switch|typedef|uchar|uchar16|uchar2|uchar3|uchar4|uchar8|uint|uint16|uint2|uint3|uint4|uint8|uintptr_t|ulong|ulong16|ulong2|ulong3|ulong4|ulong8|uniform|union|unsigned|ushort|ushort16|ushort2|ushort3|ushort4|ushort8|void|volatile|while|write_only)\b/,
```

Comment:
Wow those are huge regexps. You might want to consider at least combining some of them?

-  `bool(16|2|3|4|8)?`
- `double(16(x(1|16|2|4|8))?|1x(1|16|2|4|8)|2(x(1|16|2|4|8))?|3|4(x(1|16|2|4|8))?|8(x(1|16|2|4|8))?)?`
- `image(1d_(array_|buffer_)?t|2d_(array_(depth_|msaa_depth_|msaa_)?|depth_|msaa_depth_|msaa_)?t|3d_t)`
etc.